content/blog/python-for-pac/index.md

```diff
@@ -0,0 +1,190 @@
+
+s3_no_public_read = ResourceValidationPolicy(
+    name="s3-no-public-read",
+    description="Prohibits setting the publicRead or publicReadWrite permission on AWS S3 buckets.",
+    validate=s3_no_public_read_validator,
+)
+
+PolicyPack(
+    name="aws-python",
+    enforcement_level=EnforcementLevel.MANDATORY,
+    policies=[
+        s3_no_public_read,
+    ],
+)
+```
+
+{{% /choosable %}}
+{{% choosable cloud azure %}}
+
+```python
+from pulumi_policy import (
+    EnforcementLevel,
+    PolicyPack,
+    ReportViolation,
+    ResourceValidationArgs,
+    ResourceValidationPolicy,
+)
+
+def storage_container_no_public_read_validator(args: ResourceValidationArgs, report_violation: ReportViolation):
+    if args.resource_type == "azure:storage/container:Container" and "containerAccessType" in args.props:
+        access_type = args.props["containerAccessType"]
+        if access_type == "blob" or access_type == "container":
+            report_violation(
+                "Azure Storage Container must not have blob or container access set. " +
+                "Read more about read access here: " +
+                "https://docs.microsoft.com/en-us/azure/storage/blobs/storage-manage-access-to-resources")
+
+storage_container_no_public_read = ResourceValidationPolicy(
+    name="storage-container-no-public-read",
+    description="Prohibits setting the public permission on Azure Storage Blob Containers.",
+    validate=storage_container_no_public_read_validator,
+)
+
+PolicyPack(
+    name="azure-python",
+    enforcement_level=EnforcementLevel.MANDATORY,
+    policies=[
+        storage_container_no_public_read,
+    ],
+)
+```
+
+{{% /choosable %}}
+{{% choosable cloud gcp %}}
+
+```python
+from pulumi_policy import (
+    EnforcementLevel,
+    PolicyPack,
+    ReportViolation,
+    ResourceValidationArgs,
+    ResourceValidationPolicy,
+)
+
+def storage_bucket_no_public_read_validator(args: ResourceValidationArgs, report_violation: ReportViolation):
+    if args.resource_type == "gcp:storage/bucketACL:BucketACL" and "predefinedAcl" in args.props:
+        acl = args.props["predefinedAcl"]
+        if acl == "public-read" or acl == "public-read-write":
+            report_violation("Storage buckets acl cannot be set to public-read or public-read-write.")
+
+storage_bucket_no_public_read = ResourceValidationPolicy(
+    name="storage-bucket-no-public-read",
+    description="Prohibits setting the publicRead or publicReadWrite permission on GCP Storage buckets.",
+    validate=storage_bucket_no_public_read_validator,
+)
+
+PolicyPack(
+    name="gcp-python",
+    enforcement_level=EnforcementLevel.MANDATORY,
+    policies=[
+        storage_bucket_no_public_read,
+    ],
+)
+```
+
+{{% /choosable %}}
+{{< /chooser >}}
+
+In the case of this Kubernetes policy example, the policy does not allow deploying a loadbalancer which would open the kubernetes cluster to the public Internet.
```

Comment:
Ah, doesn't look like there's a way to use "k8s" or "kubernetes" with `{{< chooser cloud "aws,azure,gcp" >}}`. @cnunciato, is that by-design?

Reply:
@justinvp Totally an oversight! You should be able to make this work by adding a few lines to `chooser.tsx`:

```diff
 --- a/components/src/components/chooser/chooser.tsx
+++ b/components/src/components/chooser/chooser.tsx
@@ -6,7 +6,7 @@ import { setLanguage, setK8sLanguage, setOS, setCloud } from "../../store/action
 export type LanguageKey = "javascript" | "typescript" | "python" | "go" | "csharp" | "fsharp" | "visualbasic"
 export type K8sLanguageKey = "typescript" | "yaml" | "typescript-kx"
 export type OSKey = "macos" | "linux" | "windows"
-export type CloudKey = "aws" | "azure" | "gcp"
+export type CloudKey = "aws" | "azure" | "gcp" | "kubernetes";

 export type ChooserMode = "local" | "global";
 export type ChooserOptionStyle = "tabbed" | "none";
@@ -408,5 +408,10 @@ export class Chooser {
             name: "GCP",
             preview: false,
         },
+        {
+            key: "kubernetes",
+            name: "Kubernetes",
+            preview: false,
+        },
     ];
 }```

Reply:
added k8s to chooser